tests/fixtures/mq_fixtures.py

```diff
@@ -25,13 +25,14 @@ def mq_create_or_update_host(flask_app, event_producer_mock):
     def _mq_create_or_update_host(
         host_data,
         platform_metadata=None,
+        overwrite_identity=True,
         return_all_data=False,
         event_producer=event_producer_mock,
         message_operation=add_host,
     ):
         if not platform_metadata:
             platform_metadata = get_platform_metadata()
-        else:
+        elif overwrite_identity:
```

Comment:
Wondering if there's a more elegant solution... do other tests break if you just check the identity field like this?
```suggestion
        elif "b64_identity" not in platform_metadata:
```

Reply:
Actually, one of the test I wrote here is meant to not have the `b64_identity` so that we can validate the custom logic :slightly_smiling_face:

Reply:
I see... I was a bit confused bc I thought we already tested with that scenario. Should this test be merged with the existing [test_rhsm_reporter_and_no_identity](https://github.com/RedHatInsights/insights-host-inventory/blob/cfa63ab04980d2fd638ac41a75fbe754999892f6/tests/test_mq_service.py#L1197) one?

It looks like that test has a different approach to creating the host without setting the Identity

Reply:
@tahmidefaz I don't remember the context for adding the `else platform_metadata["b64_identity"] = get_encoded_idstr()` statement.  If you look at [PR-856](https://github.com/RedHatInsights/insights-host-inventory/pull/856), this `else` statement has been taken out.  I run into the same issue while writing a test involving 'CN` change.  If you take out the `else` statement before your change, `overwrite_identity` parameter is not needed anymore.

Reply:
@thearifismail great suggestion! Removing that `else` statement did fail one of the tests, but it was very simple to fix by just adding a `reporter` and `subscription_manager_id`. So, I went ahead and removed it!

@kruai I have been poking around and trying the follow the pattern in that test and I think it was getting a little complex and confusing. I choose to simply take Arif's suggestion and keep this test simple. Lmk, if you think otherwise :slightly_smiling_face:

Reply:
Yep, that works for me. I just didn't want to add yet another param to that function when it only seemed to be used in that one test. I like the way it looks now